web/src/main/java/org/mskcc/cbio/oncokb/api/pub/v1/VariantsApiController.java

```diff
@@ -34,67 +37,67 @@
         , @ApiParam(value = "variant name.") @RequestParam(value = "variant", required = false) String variant
         , @ApiParam(value = "") @RequestParam(value = "variantType", required = false) String variantType
         , @ApiParam(value = "") @RequestParam(value = "consequence", required = false) String consequence
-//        , @ApiParam(value = "") @RequestParam(value = "refResidues", required = false) String refResidues
         , @ApiParam(value = "") @RequestParam(value = "proteinStart", required = false) Integer proteinStart
         , @ApiParam(value = "") @RequestParam(value = "proteinEnd", required = false) Integer proteinEnd
-//        , @ApiParam(value = "") @RequestParam(value = "variantResidues", required = false) String variantResidues
     ) {
-        HttpStatus httpStatus = HttpStatus.OK;
+        VariantSearchQuery query = new VariantSearchQuery(entrezGeneId, hugoSymbol, variant, variantType, consequence, proteinStart, proteinEnd);
+        return new ResponseEntity<>(getVariants(query), HttpStatus.OK);
+    }
+
+    @Override
+    public ResponseEntity<List<List<Alteration>>> variantsLookupPost(@ApiParam(value = "List of queries.", required = true) @RequestBody(required = true) List<VariantSearchQuery> body) {
+        List<List<Alteration>> result = new ArrayList<>();
+        if (body != null) {
+            for (VariantSearchQuery query : body) {
+                result.add(getVariants(query));
+            }
+        }
+        return new ResponseEntity<>(result, HttpStatus.OK);
+    }
+
+    private List<Alteration> getVariants(VariantSearchQuery query) {
+        LinkedHashSet<Alteration> alterationSet = new LinkedHashSet<>();
         List<Alteration> alterationList = new ArrayList<>();
-        if (hugoSymbol != null || entrezGeneId != null) {
-            Gene gene = GeneUtils.getGene(entrezGeneId, hugoSymbol);
-            if (gene != null) {
-                Set<Alteration> allAlterations = AlterationUtils.getAllAlterations(gene);
-                if (variant == null && proteinStart == null && proteinEnd == null) {
-                    alterationList.addAll(allAlterations);
+        if (query != null) {
+            if (query.getHugoSymbol() != null || query.getEntrezGeneId() != null) {
+                Gene gene = GeneUtils.getGene(query.getEntrezGeneId(), query.getHugoSymbol());
+                if (gene != null) {
+                    if (AlterationUtils.isInferredAlterations(query.getVariant())) {
```

Comment:
No result returned for "Likely" known effect, such as Likely Oncogenic Mutations, since in the isInferredAlterations function we are using equal for known effect filter. Mutations with likely oncogenic known effect should be returned for Likely Oncogenic Mutations query.